src/reactComponents/smart-avatar.js

```diff
@@ -12,22 +12,34 @@ export default class SmartWebexTeamsAvatar extends Component {
     super(props);
 
     this.personID = props.personID;
+    this._isMounted = false;
     this.adapter = props.adapter;
     this.state = {
       person: undefined
     };
   }
 
   componentDidMount() {
-    this.adapter.getPerson(this.personID).then((person) => {
-      this.updatePerson(person);
-    })
+    this._isMounted = true;
+    this.adapter.getPerson(this.personID)
+      .subscribe(
+        person => this.updatePerson(person),
+        error => console.error(error),
+        () => {} // Subscription had been completed)
+      ); 
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
   }
 
   updatePerson(person) {
-    this.setState({
-      person
-    });
+    // Common shared adapters will even change the 
```

Comment:
I don't understand this. Seems like a workaround for something we're doing incorrectly. What caused this?